sched_boundary/sched_boundary.py

```diff
@@ -265,6 +265,8 @@ def extract_file(self, src_f):
                     line = line.replace('DEFINE_STATIC_KEY_FALSE(', 'DECLARE_STATIC_KEY_FALSE(').replace('static ', '')
                 elif 'DEFINE_STATIC_KEY_TRUE(' in line:
                     line = line.replace('DEFINE_STATIC_KEY_TRUE(', 'DECLARE_STATIC_KEY_TRUE(').replace('static ', '')
+                elif 'EXPORT_' in line and '_SYMBOL' in line:
+                    pass
```

Comment:
What about `line = “”`. Then this line gets deleted automatically, and we can get rid of the `sed` command above. This avoids us doing the same thing at many places

Reply:
No, the EXPORT_*_SYMBOL can be expanded into eight declarations in kernel 5.10，but these declarations are not needed, so I delete them. If use line = "", the first declaration can be deleted not all of them.

Reply:
I still don’t get it. With this line, all declarations (definitions in GCC plugin’s eyes because of the assembly form as you explained in your commit log) expanded by `EXPORT_TRACEPOINT_SYMBOL` will be removed from `var_list`.

After that, none of these 8 declarations will be generated in the form of `extern const char ***`. If they are not generated at all, why do we need to delete them one by one ?

Reply:
These eight decls have the same row_start and row_end, if lines[row_start]="" at the first time, other decls can not be matched and recognized.

Reply:
So we modifies the source code too pre-maturely, try something like this,

```
orig_lines = copy.deepcopy(lines)
for decl, row_start, row_end in list(self.var_list):
    line = orig_lines[row_start]
    if 'DEFINE_PER_CPU(' in line:
        line = line.replace('DEFINE_PER_CPU(', 'DECLARE_PER_CPU(').replace('static ', '')
    elif 'DEFINE_PER_CPU_SHARED_ALIGNED(' in line:
        line = line.replace('DEFINE_PER_CPU_SHARED_ALIGNED(', 'DECLARE_PER_CPU_SHARED_ALIGNED(').replace('static ', '')
    elif 'DEFINE_STATIC_KEY_FALSE(' in line:
        line = line.replace('DEFINE_STATIC_KEY_FALSE(', 'DECLARE_STATIC_KEY_FALSE(').replace('static ', '')
    elif 'DEFINE_STATIC_KEY_TRUE(' in line:
        line = line.replace('DEFINE_STATIC_KEY_TRUE(', 'DECLARE_STATIC_KEY_TRUE(').replace('static ', '')
    else:
        lines[row_start] = ''
        continue
    lines[row_start] = line
    self.var_list.remove((decl, row_start, row_end))
```

Reply:
OK, this way has an advantage that is the EXPORT_*_SYMBOL can be deleted by sched_boundary.py and cli.py dose not need sed command to delete them again.